transform/operations_test.go

```diff
@@ -45,31 +46,55 @@ type opTests struct {
 func runOpTests(t *testing.T, opType func() transformOperation, tests []opTests) {
 
 	for _, test := range tests {
-		op := opType()
-		err := op.init(test.args)
+		runOpTest(t, opType, test)
+	}
+}
 
-		switch {
-		case test.wantInitErr && err != nil:
-			continue
-		case test.wantInitErr && err == nil:
-			t.Errorf("Test %q - got init error nil, want error", test.description)
-		case !test.wantErr && err != nil:
-			t.Errorf("Test %q - got init error, want nil: %v", test.description, err)
-		}
+func runOpTest(t *testing.T, opType func() transformOperation, test opTests) {
+	op, err := runOpTestInit(opType, test)
+	if test.wantInitErr && err != nil {
+		return
+	} else if err != nil {
+		t.Error(err)
+	}
+
+	_, err = runOpTestTransform(op, test)
+	if test.wantErr && err != nil {
+		return
+	} else if err != nil {
+		t.Error(err)
+	}
+}
 
-		got, err := op.transform(test.in)
+func runOpTestInit(opType func() transformOperation, test opTests) (transformOperation, error) {
+	op := opType()
+	err := op.init(test.args)
+	switch {
+	case test.wantInitErr && err != nil:
+		return op, err
+	case test.wantInitErr && err == nil:
+		return op, fmt.Errorf("Test %q - got init error nil, want error", test.description)
+	case !test.wantErr && err != nil:
```

Comment:
I feel like there's a lot of duplicated code in the error vs wantErr checking. It can be pulled out pretty easily into a new method that accepts two `error`s as arguments and returns an `error`